staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go

```diff
@@ -119,15 +121,14 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			// the work needs to send to it. This is defer'd to ensure it runs
 			// ever if the post timeout work itself panics.
 			go func() {
+				timedOutAt := time.Now()
 				res := <-resultCh
-				if res != nil {
-					switch t := res.(type) {
-					case error:
-						utilruntime.HandleError(t)
-					default:
-						utilruntime.HandleError(fmt.Errorf("%v", res))
-					}
-				}
+
+				// a non empty reason indicates that there was a panic.
+				metrics.RecordRequestPostTimeout("timeout-handler", strconv.FormatBool(res != nil), strconv.FormatBool(false))
```

Comment:
If we're going to be storing bools here, we shouldn't be storing the string 'unknown' in the same field..

Reply:
On second thought, I collapsed both into one label `reason`, possible values: `panicd`, `error`, `ok` and `unknown`

Reply:
isn't a panic just a type of error?

Reply:
@logicalhan  I think it's useful to preserve the distinction (or context) here - the metric can answer the question whether the child goroutine executing `ResultFunc` panicked or returned an `error`.  let me know if you have any concern with the cardinality here. 

Also, this counter is not of O(number of requests in flight), but is of O(request timed out).

Reply:
okay, then i would use 'panic' instead of 'panicd'

Reply:
@logicalhan done, PTAL